lib/withOnyx.js

```diff
@@ -37,12 +37,27 @@ export default function (mapOnyxToState) {
                 // disconnected. It is a key value store with the format {[mapping.key]: connectionID}.
                 this.activeConnectionIDs = {};
 
+                const cachedState = {};
+
+                // Check if we can get all the data synchronously from cache
+                _.each(mapOnyxToState, (mapping, propertyName) => {
+                    const key = Str.result(mapping.key, props);
+                    const value = Onyx.tryGetCachedValue(key, mapping);
+
+                    if (value !== undefined) {
+                        cachedState[propertyName] = value;
+                    }
+                });
+
+                // If we have all the data we need, then we can render the component immediately
+                const loading = _.keys(cachedState).length < requiredKeysForInit.length;
```

Comment:
I think we only need to set the `cachedState` if we are still loading stuff.

Reply:
Hm, not sure i understand you correctly. With the current changes we 

- always set all the cached state we could load for the keys to the component state 
- if not all keys have been retrievable from cache the loading value will be true

Reply:
Sorry that was really unclear haha. 

I meant that setting up the `this.tempState` variable is not needed if we got the entire `cachedState`? Because we would never use it for anything.

(not sure why my brain said the other thing)

Reply:
Oh, we continue using the tempState later on in `setStateWithOnyx`

Reply:
Only if `loading: true` ?

Reply:
yeah, as `loading: true` is like "we didn't have all the data in cache and need to wait for storage or maybe even wait until a network request puts the data into onyx".